ocaml-lsp-server/src/diagnostics.ml

```diff
@@ -19,6 +19,45 @@ end
 
 module Dune = Stdune.Id.Make ()
 
+let equal_message =
+  (* because the compiler and merlin wrap messages differently *)
+  let is_space = function
+    | ' '
+    | '\r'
+    | '\n'
+    | '\t' ->
+      true
+    | _ -> false
+  in
+  let eat_space s i =
+    while !i < String.length s && is_space s.[!i] do
+      incr i
+    done
+  in
+  fun m1 m2 ->
+    let i = ref 0 in
+    let j = ref 0 in
+    try
+      eat_space m1 i;
+      eat_space m2 j;
+      while !i < String.length m1 && !j < String.length m2 do
+        let ci = m1.[!i] in
+        let cj = m2.[!j] in
+        eat_space m1 i;
+        eat_space m2 j;
+        if (is_space ci && is_space cj) || Char.equal ci cj then (
+          incr i;
+          incr j
```

Comment:
when we eat all space, we set `i` and `j` to non-space chars, but then we increment over them because in the if condition we check `ci` and `cj` which are values at old `i` and `j` index values. As a result, 

```ocaml
utop # equal_message "foo bar" "foo gar";;
- : bool = true
```

Reply:
Fixed. Thanks.